rest/replicator_test.go

```diff
@@ -2445,6 +2445,121 @@ func TestActiveReplicatorRecoverFromMismatchedRev(t *testing.T) {
 	assert.NoError(t, ar.Stop())
 }
 
+// TestActiveReplicatorReconnectOnStart ensures ActiveReplicators retry their initial connection for cases like:
+// - Incorrect credentials
+// - Unroutable remote address
+func TestActiveReplicatorReconnectOnStart(t *testing.T) {
+
+	if base.GTestBucketPool.NumUsableBuckets() < 2 {
+		t.Skipf("test requires at least 2 usable test buckets")
+	}
+
+	tests := []struct {
+		name                  string
+		usernameOverride      string
+		remoteURLHostOverride string
+		expectedErrorContains string
+	}{
+		{
+			name:                  "wrong user",
+			usernameOverride:      "bob",
+			expectedErrorContains: "unexpected status code 401 from target database",
+		},
+		{
+			name:                  "invalid port", // fails faster than unroutable address (connection refused vs. connect timeout)
+			remoteURLHostOverride: "127.0.0.1:1234",
+			expectedErrorContains: "connection refused",
+		},
```

Comment:
As discussed, could add a successful retry case by starting the replication, then creating the user on the remote after it enters the retry loop.